geomstats/geometry/base.py

```diff
@@ -324,24 +324,23 @@ class LevelSet(Manifold, abc.ABC):
     """
 
     def __init__(
-        self,
-        dim,
-        embedding_space,
-        submersion,
-        value,
-        tangent_submersion,
-        default_coords_type="intrinsic",
-        **kwargs
+        self, dim, embedding_space, value, default_coords_type="intrinsic", **kwargs
     ):
         kwargs.setdefault("shape", embedding_space.shape)
         super().__init__(dim=dim, default_coords_type=default_coords_type, **kwargs)
         self.embedding_space = embedding_space
         self.embedding_metric = embedding_space.metric
-        self.submersion = submersion
         if isinstance(value, float):
             value = gs.array(value)
         self.value = value
-        self.tangent_submersion = tangent_submersion
+
+    @abc.abstractmethod
+    def submersion(self, point):
+        pass
```

Comment:
Do you need `pass` ? I thought abstract methods can be empty. However, it might be better to keep pass so that child classes do not necessarily have to provide these? (see my last comment on the related github issue).

Reply:
I don't know the rule for abc.abstractmethod

Reply:
After adding docstrings it will be possible to have no code, but not without.